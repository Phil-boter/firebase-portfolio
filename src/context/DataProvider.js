import React, { useEffect, useState} from "react";


export const DataContext = React.createContext();

export function DataProvider({ children }) {
    const [projectsData, setCurrentProjectData] = useState();
    const [loading, setLoading] = useState(true);

    async function getProjects() {
            let projects = [];
            fetch("http://localhost:3500/v1/projects/allProjects")
            .then((response) => {
                if (!response.ok) {
                  throw new Error(
                    `This is an HTTP error: The status is ${response.status}`
                  );
                }
                return response.json();
              })
              .then((data)=> {
                console.log("data", data)
                    projects = data.rows;
                    setCurrentProjectData(projects);
                    setLoading(false);
                
        })
    }

    useEffect(() => {
        console.log("use effect")
        getProjects();
    }, []);

    const value = {
        projectsData,
    };

    return (
        <DataContext.Provider value={value}>
            {!loading && children}
        </DataContext.Provider>
    );
}
