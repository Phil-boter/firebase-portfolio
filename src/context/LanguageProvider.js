import React, { createContext, useContext, useState } from "react";

export const LanguageContext = createContext(null);
