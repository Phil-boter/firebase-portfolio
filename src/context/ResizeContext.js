import { useEffect, useState, createContext } from "react";

export const ResizeContext = createContext();

function debounce(fn, ms) {
    let timer;
    return (cleanUp) => {
        clearTimeout(timer);
        timer = setTimeout((_) => {
            timer = null;
            fn.apply(this, arguments);
        }, ms);
    };
}

export function ResizeProvider({ children }) {
    const [dimensions, setDimensions] = useState({
        height: window.innerHeight,
        width: window.innerWidth,
    });

    console.log("dim in provider", dimensions);

    useEffect(() => {
        const debouncedHandleResize = debounce(function handleResize() {
            setDimensions({
                height: window.innerHeight,
                width: window.innerWidth,
            });
        }, 500);

        window.addEventListener("resize", debouncedHandleResize);

        return function cleanUp() {
            window.removeEventListener("resize", debouncedHandleResize);
        };
    }, [dimensions, setDimensions]);

    const value = { dimensions };

    return (
        <ResizeContext.Provider value={value}>
            {children}
        </ResizeContext.Provider>
    );
}
