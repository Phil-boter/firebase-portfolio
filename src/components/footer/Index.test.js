import React from "react";

import { MemoryRouter } from "react-router-dom";

import { render, fireEvent, cleanup } from "@testing-library/react";

import { LanguageContext } from "../../context/LanguageProvider";
import Footer from "./Index";

let location = {
    hash: "",
    path: "/",
    search: "",
    state: undefined,
    key: "x774kd",
};
let language = "en";

afterEach(cleanup);

describe("Footer component", () => {
    it("should render Footer and match snapshot", () => {
        const { container } = render(
            <MemoryRouter location={location}>
                <LanguageContext.Provider value={{ language }}>
                    <Footer language={language} />
                </LanguageContext.Provider>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it("should have an container with a class of footer-article", () => {
        const { container } = render(
            <MemoryRouter location={location}>
                <LanguageContext.Provider value={{ language }}>
                    <Footer language={language} />
                </LanguageContext.Provider>
            </MemoryRouter>
        );
        expect(container.querySelector(".footer-article")).toBeInTheDocument();
    });

    it("should render button, that redirrct to about if Click", async () => {
        const clickToSpy = jest.fn();
        global.open = clickToSpy;
        const { container } = render(
            <MemoryRouter location={location}>
                <LanguageContext.Provider value={{ language }}>
                    <Footer language={language} />
                </LanguageContext.Provider>
            </MemoryRouter>
        );

        fireEvent.click(container.querySelector(".GIT"));
        expect(clickToSpy).toHaveBeenCalled();
        expect(window.open).toBeCalledWith("https://github.com/Phil-boter");

        fireEvent.click(container.querySelector(".LINKEDIN"));
        expect(clickToSpy).toHaveBeenCalled();
        expect(window.open).toBeCalledWith(
            "https://www.linkedin.com/in/philipp-dawid-759793206/"
        );
    });
});
