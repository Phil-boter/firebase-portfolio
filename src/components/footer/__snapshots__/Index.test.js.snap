// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render Footer and match snapshot 1`] = `
<div>
  <div
    class="email-modal"
  >
    <div
      class="email-main-wrapper"
    >
      <div
        class="contact-form-main-container"
      >
        <div
          class="contact-form-header main-header"
        >
          <h1>
            GET IN CONTACT
          </h1>
        </div>
        <div
          class="contact-form-container"
          id="contact-form-container"
        >
          <div
            class="contact-form-inner-container"
            id="emailSlider"
          >
            <form
              class="contact-form"
              id="reset"
            >
              <label>
                Name
              </label>
              <input
                autocomplete="on"
                class="contact-form-input"
                id="imput-field-name"
                name="name"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
              <label>
                Email
              </label>
              <input
                autocomplete="on"
                class="contact-form-input"
                id="imput-field-email"
                name="email"
                placeholder="Email"
                required=""
                type="email"
                value=""
              />
              <label>
                Title
              </label>
              <input
                class="contact-form-input"
                id="input-field-title"
                name="Title"
                placeholder="Title"
                required=""
                type="text"
                value=""
              />
              <label>
                Message
              </label>
              <textarea
                class="contact-form-input contact-form-textarea"
                id="input-textarea"
                name="message"
                placeholder="Write your message here"
                required=""
                rows="9"
              />
              <button
                class="button-contact-form send"
              >
                Submit
              </button>
              <button
                class="button-contact-form"
              >
                Reset
              </button>
              <label
                class="ohnohoney"
              />
              <input
                autocomplete="off"
                class="ohnohoney"
                id="input-name-honeypot"
                name="name"
                placeholder="Your name here"
                type="text"
              />
              <label
                class="ohnohoney"
              />
              <input
                autocomplete="off"
                class="ohnohoney"
                id="input-email-honeypot"
                name="email"
                placeholder="Your e-mail here"
                type="emely"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="footer-container"
  >
    <article
      class="footer-article"
    >
      <span
        class="link-linkedIn GIT"
      >
        <img
          alt="github"
          class="arrow slide-footer"
          src="/assets/git.png"
        />
      </span>
      <span
        class="link-linkedIn LINKEDIN"
      >
        <img
          alt="linkedIn"
          class="arrow slide-footer"
          src="/assets/linkedIn.png"
        />
      </span>
      <span
        class="link-linkedIn MAILTO"
      >
        <img
          alt="mail"
          class="arrow slide-footer"
          src="/assets/mail.png"
        />
      </span>
    </article>
    <a
      href="/contact"
    >
      <button
        class="button"
      >
        Get in Contact
      </button>
    </a>
  </section>
</div>
`;
